src/Uno.UI.Tests/Windows_UI_Xaml/Given_StaticResource.cs

```diff
@@ -19,9 +25,452 @@ public void When_Resource_In_Application_Merged()
+			var text1ResourceTemplateAfter = control.TemplateFromResourceControl.TextBlock1.Text;
+
+			Assert.AreEqual("LocalVisualTree", text1InlineBefore);
+			Assert.AreEqual("OuterVisualTree", text1ResourceTemplateBefore);
+
+			Assert.AreEqual("ApplicationLevel", text3InlineBefore);
+			Assert.AreEqual("ApplicationLevel", text3ResourceTemplateBefore);
+
+			Assert.AreEqual("LocalVisualTree", text1InlineAfter);
+			Assert.AreEqual("OuterVisualTree", text1ResourceTemplateAfter);
+		}
+
+		[TestMethod]
+		public void When_Multiple_References_Framework_DependencyObject()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			app.HostView.Children.Add(control);
+
+			control.Measure(new Size(1000, 1000));
+
+			var foreground = control.InlineTemplateControl.TextBlock1.Foreground as SolidColorBrush;
+			var background = control.TemplateFromResourceControl.Background as SolidColorBrush;
+			var inResources = control.Resources["MutatedColorBrush"] as SolidColorBrush;
+
+			Assert.AreEqual(Colors.Chartreuse, foreground.Color);
+			Assert.AreEqual(Colors.Chartreuse, background.Color);
+			Assert.AreEqual(Colors.Chartreuse, inResources.Color);
+
+			Assert.IsTrue(ReferenceEquals(foreground, background));
+			Assert.IsTrue(ReferenceEquals(inResources, foreground));
+
+			inResources.Color = Colors.GhostWhite;
+			Assert.AreEqual(Colors.GhostWhite, foreground.Color);
+			Assert.AreEqual(Colors.GhostWhite, background.Color);
+
+			var control2 = new Test_Control();
+			app.HostView.Children.Add(control2);
+			control2.Measure(new Size(1000, 1000));
+
+
+			var background2 = control2.TemplateFromResourceControl.Background as SolidColorBrush;
+			var areRefEqual = ReferenceEquals(background, background2);
+			Assert.IsFalse(areRefEqual);
+			Assert.AreEqual(Colors.Chartreuse, background2.Color);
+		}
+
+		[TestMethod]
+		public void When_Multiple_References_Poco()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			var poco1 = control.InlineTemplateControl.Poco;
+			var poco2 = control.TemplateFromResourceControl.Poco;
+			Assert.IsTrue(ReferenceEquals(poco1, poco2));
+
+			Assert.AreEqual(101, poco1.Bogosity);
+			Assert.AreEqual(101, poco2.Bogosity);
+
+			poco1.Bogosity = 107;
+			Assert.AreEqual(107, poco2.Bogosity);
+
+			var control2 = new Test_Control();
+			var poco3 = control2.InlineTemplateControl.Poco;
+			Assert.IsFalse(ReferenceEquals(poco1, poco3));
+			Assert.AreEqual(101, poco3.Bogosity);
+		}
+
+		[TestMethod]
+		public void When_Multiple_References_Custom_DependencyObject()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			var do1_1 = control.InlineTemplateControl.DObjDP;
+			var do1_2 = control.InlineTemplateControl.PlainDO;
+			var do1_3 = control.TemplateFromResourceControl.DObjDP;
+			var areRefEqual1 = ReferenceEquals(do1_1, do1_2);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'areRefEqual1'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)